frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -419,32 +418,54 @@ export const fetchCardData = createThunkAction(
   },
 );
 
-export const fetchDashboardCardData = createThunkAction(
-  FETCH_DASHBOARD_CARD_DATA,
-  options => (dispatch, getState) => {
+const fetchDashboardCardDataSync = createAction(FETCH_DASHBOARD_CARD_DATA);
+
+export const fetchDashboardCardData =
+  ({ skipOngoingDashcardRequests, ...options }) =>
+  (dispatch, getState) => {
     const dashboard = getDashboardComplete(getState());
     const selectedTabId = getSelectedTabId(getState());
-    const dashcardIds = [];
-    const promises = getCurrentTabDashboardCards(dashboard, selectedTabId)
+
+    const nonVirtualDashcardsToFetch = getCurrentTabDashboardCards(
+      dashboard,
+      selectedTabId,
+    )
       .filter(({ dashcard }) => !isVirtualDashCard(dashcard))
-      .map(({ card, dashcard }) => {
-        dashcardIds.push(dashcard.id);
-        return dispatch(fetchCardData(card, dashcard, options)).then(() => {
-          return dispatch(updateLoadingTitle());
-        });
+      .filter(({ dashcard }) => {
+        if (skipOngoingDashcardRequests) {
+          const loadingIds = getLoadingDashCards(getState()).loadingIds;
```

Comment:
I deliberately put `loadingIds` here, so it's closer to where it's used. I know calling a function every loop might not be the best practice generally, but in this case, it should be negligible since `getLoadingDashCards` is just getting a value from our redux store, and doesn't involve any expensive calculations.